app/lib/analyzer/task_sources.dart

```diff
@@ -77,14 +77,17 @@ class DatastoreHistoryTaskSource implements TaskSource {
   Stream<Task> startStreaming() async* {
     for (;;) {
       try {
-        await for (PackageVersion pv in _db.query(PackageVersion).run()) {
-          await new Future.delayed(const Duration(seconds: 30));
+        final Query query = _db.query(PackageVersion)..order('-created');
+        await for (PackageVersion pv in query.run()) {
           final List<PackageVersionAnalysis> list = await _db.lookup([
             _db.emptyKey
                 .append(PackageAnalysis, id: pv.package)
                 .append(PackageVersionAnalysis, id: pv.version)
           ]);
           if (list.first == null) {
+            // Waiting to throttle the historical tasks. If there is a newly
+            // uploaded package, give it a chance before continuing with these.
+            await new Future.delayed(const Duration(seconds: 30));
```

Comment:
The fact that the comment is explaining things about "newly uploaded package" is a clear sign that it's code which doesn't fit in there (the only responsibility of the class should be scanning historical packages, it shouldn't care about how many task sources the user of the class has and how the tasks are prioritized).

But if you want to move that functionality in the stream merging in the future, it's fine with me.